compiler/optimizer/Inliner.cpp

```diff
@@ -3265,13 +3265,13 @@ TR::TreeTop * OMR_InlinerUtil::storeValueInATemp(
       bool isInternalPointer = false;
 
       // disable internal pointer symbols for C/C++
-      if (
-         (value->getOpCode().isArrayRef() ||
+      if (value->canBeInternalPtrOfObject() ||
             (value->getOpCode().isLoadVarDirect() &&
              value->getSymbolReference()->getSymbol()->isAuto() &&
-             value->getSymbolReference()->getSymbol()->castToAutoSymbol()->isInternalPointer())))
+             value->getSymbolReference()->getSymbol()->castToAutoSymbol()->isInternalPointer()))
          isInternalPointer = true;
 
+      // value->isNotCollected() == false does not mean value is collected
```

Comment:
Not sure what you are trying to tell us here... perhaps elaborate about what we are guarding and why the check is the way it is?